### PR TITLE
=pro disable parallel test execution to make tests more reliable

### PR DESCRIPTION
**Pull request by jrudolph**

### CODE REVIEW

**Comment by jrudolph** [comment]
I'm still experimenting with the settings... The ones that I thought should work, don't.

**Comment by jrudolph** [comment]
PLS BUILD ALL

**Comment by jrudolph** [comment]
Approximately doubles the full execution time of all tests. I guess that time is well-spent for the nightlies and may still be good enough for PR validation. Full time for PR validation would afterwards be ~ 8 minutes.

File: project/ValidatePullRequest.scala [hunk 1]
```diff
@@ -84,6 +84,9 @@ object ValidatePullRequest extends AutoPlugin {
   val validatePullRequest = taskKey[Unit]("Validate pull request")
   val additionalTasks = taskKey[Seq[TaskKey[_]]]("Additional tasks for pull request validation")
 
+  // The set of (top-level) files or directories to watch for build changes.
+  val BuildFilesAndDirectories = Set("project", "build.sbt")
```
Comment (ktoso) [inline project/ValidatePullRequest.scala]: good catch

File: build.sbt [hunk 2]
```diff
@@ -44,7 +44,8 @@ inThisBuild(Def.settings(
   testOptions += Tests.Argument(TestFrameworks.JUnit, "-q", "-v"),
   Dependencies.Versions,
   Formatting.formatSettings,
-  shellPrompt := { s => Project.extract(s).currentProject.id + " > " }
+  shellPrompt := { s => Project.extract(s).currentProject.id + " > " },
+  concurrentRestrictions in Global += Tags.limit(Tags.Test, 1)
```
Comment (ktoso) [inline build.sbt]: yeah sounds good, until we stabilize things; weird that it started failing so much on our infra recently after the upgrades

Comment (jrudolph) [reply build.sbt]: No, it runs only on my machine right now. I guess the much bigger amount of tests after the client pool merges changed the execution order in a way that makes it more likely somehow.

**Pull request merged**